sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -247,44 +247,36 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           _, left, right, hint) =>
         val hashJoinSupport = hashJoinSupported(leftKeys, rightKeys)
```

Comment:
I think that you still need to keep hashJoinSupport flag check. E.g. if physical plan is manually constructed you don't have any guarantee that CollationKey QO rule kicked in.

I would prefer collation key returning binary normalized value that does support hash join and keep hashJoinSupport check for StringType.

Also, in future, we may want to add rules in QO that will chose to avoid collation key path, given that that path is also not cheap and not clearly better than sort merge join (we need to measure this).

Reply:
this makes sense, will add these checks back